src/language-js/comments.js

```diff
@@ -942,6 +947,46 @@ function handleTSMappedTypeComments(
   return false;
 }
 
+function hasQuestionBetweenTestAndComment(testNode, comment, text, options) {
+  const testNodeLocEnd = options.locEnd(testNode);
+  const commentLocStart = options.locStart(comment);
+  for (let i = testNodeLocEnd; i < commentLocStart; i++) {
+    const idx = privateUtil.getNextNonSpaceNonCommentCharacterIndexWithStartIndex(
+      text,
+      i
+    );
+    if (text[idx] === "?") {
+      return idx < commentLocStart;
+    }
+  }
```

Comment:
```suggestion
  for (let i = testNodeLocEnd; i < commentLocStart; ) {
    i = privateUtil.getNextNonSpaceNonCommentCharacterIndexWithStartIndex(
      text,
      i
    );
    if (text[i] === "?") {
      return i < commentLocStart;
    }
  }
```

This work?

Reply:
No, the loop never ends for below code:

```js
(test) ?
  // comment
  first
  : second;
```

Reply:
Why? In this case `testNodeLocEnd` is 5, `commentLocStart` is 11, 

1st loop, `i` will be 6(index of `)`),
2nd loop, `i` will be 8(index of `?`), and exit

Reply:
`testNodeLocEnd` is `4`, `commentLocStart` is `10`. Why `i` will be increased? `getNextNonSpaceNonCommentCharacterIndexWithStartIndex` always return `4`(`text[4]` is `)`).

Reply:
`getNextNonSpaceNonCommentCharacterIndexWithStartIndex(text, i)`

The second argument, startIndex increased, the result will increase.

Am I wrong?

Reply:
Hmm, I'm not sure. At least it does not work on my local. I might misunderstand, please pull and try.

Reply:
Ah, I see the problem, `getNextNonSpaceNonCommentCharacterIndexWithStartIndex(text, index)` returns `index`, if `text[index]` is valid character.

So `+1` should work, I'll try on my machine.